src/hmem_cuda.c

```diff
@@ -314,7 +314,10 @@ int cuda_copy_from_dev(uint64_t device, void *dst, const void *src, size_t size)
 
 int cuda_dev_register(const void *addr, size_t size, uint64_t *handle)
 {
-	return cuda_gdrcopy_dev_register(addr, size, handle);
+	if (cuda_is_gdrcopy_enabled()) {
+		return cuda_gdrcopy_dev_register(addr, size, handle);
+	}
+	return -FI_EIO;
```

Comment:
FI_ENOSYS makes more sense to me

Reply:
Hopefully the new commits address your change request. let me know if something else is needed at this point